File: app/src/main/java/com/johnnyzhou/movieme/di/component/FragmentComponent.java
package com.johnnyzhou.movieme.di.component;

import com.johnnyzhou.movieme.di.module.FragmentModule;
import com.johnnyzhou.movieme.di.scope.PerFragment;
import com.johnnyzhou.movieme.ui.movie.detail.MovieDetailFragment;
import com.johnnyzhou.movieme.ui.movie.list.MovieListFragment;
import com.johnnyzhou.movieme.ui.person.detail.PersonDetailFragment;
import com.johnnyzhou.movieme.ui.person.list.PersonListFragment;

import dagger.Component;

@PerFragment
@Component(dependencies = AppComponent.class, modules = FragmentModule.class)
public interface FragmentComponent {
    void inject(MovieListFragment fragment);

//    void inject(MovieDetailFragment fragment);

    void inject(PersonListFragment fragment);

//    void inject(PersonDetailFragment fragment);
}

File: app/src/main/java/com/johnnyzhou/movieme/di/component/FragmentComponent.kt
package com.johnnyzhou.movieme.di.component

import android.content.Context
import com.johnnyzhou.movieme.di.module.FragmentBindingModule
import com.johnnyzhou.movieme.di.module.FragmentModule
import com.johnnyzhou.movieme.di.scope.PerFragment
import com.johnnyzhou.movieme.ui.movie.list.MovieListFragment
import com.johnnyzhou.movieme.ui.person.list.PersonListFragment
import dagger.BindsInstance
import dagger.Component

@PerFragment
@Component(
    dependencies = [AppComponent::class],
    modules = [FragmentModule::class, FragmentBindingModule::class]
)
interface FragmentComponent {
    fun inject(fragment: MovieListFragment?)

    fun inject(fragment: PersonListFragment?)

//    @Component.Factory
//    interface Factory {
//        fun create(@BindsInstance appComponent: AppComponent): FragmentComponent
//    }
}